reports/performance.qmd

```diff
@@ -14,14 +14,2012 @@ editor: source
+  ) %>%
+  group_by(meta_township_name, Quarter, Bin) %>%
+  summarise(Sales = n()) %>%
+  ungroup() %>%
+  select(Sales, Bin, Township = meta_township_name, Quarter) %>%
+  ggplot(aes(
+    x = Quarter,
+    y = Sales,
+    fill = Bin,
+    group = Bin
+  )) +
+  geom_area() +
+  scale_color_brewer(palette = "PuOr") +
+  theme_minimal() +
+  theme(
+    legend.position = "bottom",
+    axis.text.x = element_text(angle = 90, hjust = 1),
+    axis.title.x = element_blank(),
+    axis.title.y = element_blank()
+  ) +
+  facet_wrap(vars(Township), scales = "free_y", ncol = 3)
+```
+
+## By Class and Year
+
+*NOTE: Outliers are removed*
+
+```{r class_year}
+class <- training_data %>%
+  filter(
+    !sv_is_outlier,
+    meta_triad_name == run_triad,
+    !ind_pin_is_multicard
+  ) %>%
+  mutate(meta_year = as.numeric(meta_year)) %>%
+  group_by(meta_class, meta_year) %>%
+  summarise(sales = n()) %>%
+  rename(Class = meta_class) %>%
+  pivot_wider(
+    id_cols = Class,
+    names_from = meta_year,
+    values_from = sales
+  )
+
+class %>%
+  kable() %>%
+  kable_styling() %>%
+  row_spec(unique(which(is.na(class), arr.ind = TRUE)[, 1]),
+    bold = TRUE,
+    background = "lightgrey"
+  )
+```
+
+## Removed Outliers
+
+These are outliers that were removed via the sale validation process in the ingest stage. The goal is to confirm that these were reasonable sales to remove.
+
+```{r}
+rbind(
+  training_data %>%
+    filter(
+      sv_is_outlier,
+      meta_triad_name == run_triad,
+      !ind_pin_is_multicard
+    ) %>%
+    slice_max(order_by = meta_sale_price, n = 25),
+  training_data %>%
+    filter(
+      sv_is_outlier,
+      meta_triad_name == run_triad,
+      !ind_pin_is_multicard
+    ) %>%
+    slice_min(order_by = meta_sale_price, n = 25)
+) %>%
+  datatable(rownames = FALSE)
+```
+
+## Kept Outliers
+
+These are the top and bottom 25 sales from the training data that are *still* in the sample. The goal is to gauge whether or not outlier removal is being too strict (or not strict enough).
+
+```{r}
+rbind(
+  training_data %>%
+    filter(
+      !sv_is_outlier,
+      meta_triad_name == run_triad,
+      !ind_pin_is_multicard
+    ) %>%
+    slice_max(order_by = meta_sale_price, n = 25),
+  training_data %>%
+    filter(
+      !sv_is_outlier,
+      meta_triad_name == run_triad,
+      !ind_pin_is_multicard
+    ) %>%
+    slice_min(order_by = meta_sale_price, n = 25)
+) %>%
+  datatable(rownames = FALSE)
+```
+:::
+
+
+
+## Data Quality Checks
+
+::: panel-tabset
+## Home Improvement Exemption Spot Check
+
+This tests if our Home Improvement Exemption characteristics are getting accurately updated.
+
+```{r}
+AWS_ATHENA_CONN_NOCTUA <- DBI::dbConnect(noctua::athena())
+
+noctua_options(cache_size = 10)
+
+hie_pins <- training_data %>%
+  filter(
+    hie_num_active == 1,
+    meta_triad_code == run_triad_code,
+    meta_year == as.numeric(params$year) - 1
+  ) %>%
+  pull(meta_pin) %>%
+  unique()
```

Comment:
@Damonamajor @wrridgeway Per [the latest run](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/%2Faws%2Fbatch%2Fjob/log-events/z_ci_move-existing-reports-into-reports-performance-qmd_ccao-data-model-res-avm%2Fdefault%2F0e0dca73898f4bec882437cdc44a4dc5) of this branch it looks like Fargate/EC2 runners don't have the permissions needed to query Athena. Can you two put your heads together tomorrow and figure out if there's a way to remove this code, move it to the ingest, or otherwise make it run?

If it can't be moved, then work with @jeancochrane to get the runners proper IAM permissions.